qt/widgets/common/test/MuonPeriodInfoTest.h

```diff
@@ -0,0 +1,107 @@
+                       const std::string &tag) {
+    const auto table = m_periodInfo->getTable();
+    TS_ASSERT_EQUALS(std::to_string(row + 1), table->item(row, 0)->text().toStdString()); // Period Count
+    TS_ASSERT_EQUALS(name, table->item(row, 1)->text().toStdString());
+    TS_ASSERT_EQUALS(type, table->item(row, 2)->text().toStdString());
+    TS_ASSERT_EQUALS(DAQNumber, table->item(row, 3)->text().toStdString());
+    TS_ASSERT_EQUALS(frames, table->item(row, 4)->text().toStdString());
+    TS_ASSERT_EQUALS(totalFrames, table->item(row, 5)->text().toStdString());
+    TS_ASSERT_EQUALS(counts, table->item(row, 6)->text().toStdString());
+    TS_ASSERT_EQUALS(tag, table->item(row, 7)->text().toStdString());
+  }
+
+  void test_that_the_table_is_empty_on_initialization() { TS_ASSERT_EQUALS(true, m_periodInfo->isEmpty()); }
+
+  void test_clear() {
+    auto table = m_periodInfo->getTable();
+    table->insertRow(0);
+    table->insertRow(1);
+    TS_ASSERT_EQUALS(false, m_periodInfo->isEmpty());
+    m_periodInfo->clear();
+    TS_ASSERT_EQUALS(true, m_periodInfo->isEmpty());
+    TS_ASSERT_EQUALS(0, m_periodInfo->getDAQCount());
+    TS_ASSERT_EQUALS(-1, m_periodInfo->getNumberOfSequences());
+    TS_ASSERT_EQUALS("Period Information for Run(s) ", m_periodInfo->getWidgetTitleRuns());
+  }
+
+  void test_addPeriodToTable() {
+    m_periodInfo->addPeriodToTable("state 1 dwell", "2", "10", "200", "25", "1");
+    m_periodInfo->addPeriodToTable("state 1", "1", "50", "1000", "25", "2");
+    assertRowValues(0, "state 1 dwell", "DWELL", "-", "10", "200", "-", "0001");
+    assertRowValues(1, "state 1", "DAQ", "1", "50", "1000", "25", "0010");
+    TS_ASSERT_EQUALS(2, m_periodInfo->getTable()->rowCount());
+    TS_ASSERT_EQUALS(false, m_periodInfo->isEmpty());
+  }
+
+  void test_addPeriodToTable_with_bad_tag() {
+    m_periodInfo->addPeriodToTable("state 1 dwell", "2", "10", "200", "25", "tag");
+    assertRowValues(0, "state 1 dwell", "DWELL", "-", "10", "200", "-", "Not found");
+  }
+
+  void test_DAQ_count_increases_as_expected() {
+    m_periodInfo->addPeriodToTable("state 1", "1", "10", "200", "25", "1");
+    TS_ASSERT_EQUALS(1, m_periodInfo->getDAQCount());
+    m_periodInfo->addPeriodToTable("state 1 dwell", "2", "10", "200", "25", "1");
+    TS_ASSERT_EQUALS(1, m_periodInfo->getDAQCount());
+    m_periodInfo->addPeriodToTable("state 1", "1", "10", "200", "25", "1");
+    m_periodInfo->addPeriodToTable("state 1", "1", "10", "200", "25", "1");
+    TS_ASSERT_EQUALS(3, m_periodInfo->getDAQCount());
+  }
+  void test_setNumberOfSequences_to_negative() {
+    m_periodInfo->setNumberOfSequences(-1);
+    TS_ASSERT_EQUALS("Number of period cycles not found", m_periodInfo->getNumberOfSequencesString());
+  }
+
+  void test_setNumberOfSequences() {
+    m_periodInfo->setNumberOfSequences(2);
+    TS_ASSERT_EQUALS("Run contains 2 cycles of periods", m_periodInfo->getNumberOfSequencesString());
```

Comment:
does this mean that the sequences are the number of periods?

Reply:
No they are not 2 is just a random arbitrary number I used for the test. This text format is the same as Wimbda

Reply:
ok. I just want to know what it means